runtime/near-vm-runner/src/cache.rs

```diff
@@ -292,3 +292,46 @@ pub mod wasmer1_cache {
         }
     }
 }
+
+pub fn cache_contract(
+    vm_kind: VMKind,
+    wasm_code: &ContractCode,
+    config: &VMConfig,
+    cache: Option<&dyn CompiledContractCache>,
+) -> Result<bool, VMError> {
+    if cache.is_none() {
+        return Ok(false);
+    }
+    let cache = cache.unwrap();
+    let key = get_key(wasm_code, vm_kind, config);
+    // Check if we already cached with such a key.
+    match cache.get(&(key.0).0) {
+        // If so - do not override.
+        // TODO: is it correct?
+        Ok(_) => return Ok(false),
```

Comment:
Huh, should this be `Ok(None)`? It seems at the moment we *always* hit the `return` branch.

Reply:
Right, fixed.